ios/Peepz/Peepz/Model/PeepzModel.swift

```diff
@@ -0,0 +1,91 @@
+import Foundation
+import GoogleSignIn
+import Firebase
+
+class PeepzModel: NSObject, ObservableObject {
+    @Published var isAuthenticated = false
+    @Published var users = [GalleryItemViewState]()
+
+    var fireDatabaseRef = Database.database().reference()
+
+    override init() {
+        super.init()
+        configureGoogleSignIn()
+    }
+
+    private func configureGoogleSignIn() {
+        GIDSignIn.sharedInstance().clientID = FirebaseApp.app()?.options.clientID
+        GIDSignIn.sharedInstance().delegate = self
+    }
+}
+
+extension PeepzModel: GIDSignInDelegate {
+    func sign(_ signIn: GIDSignIn!, didSignInFor user: GIDGoogleUser!, withError error: Error?) {
+
+        if let error = error {
+            print("Sign In failed with error: \(error)")
+            return
+        }
+
+        guard let authentication = user.authentication else { return }
+        let credential = GoogleAuthProvider.credential(withIDToken: authentication.idToken,
+                                                       accessToken: authentication.accessToken)
+
+        firebaseAuth(credential: credential)
+    }
+
+    func sign(_ signIn: GIDSignIn!, didDisconnectWith user: GIDGoogleUser!, withError error: Error!) {
+        // Perform any operations when the user disconnects from app here.
+        print("Logged out: Yet to be implemented")
+    }
+
+    func firebaseAuth(credential: AuthCredential) {
+        Auth.auth().signIn(with: credential) { (authResult, error) in
+
+            if let error = error {
+                print("Firebase Login failed: \(error)")
+                return
+            }
+            self.isAuthenticated = true
```

Comment:
I think firebase has a isSignedIn var or function that you may be able to listen to instead

Reply:
/** @fn isSignInWithEmailLink
        @brief Checks if link is an email sign-in link.
    
        @param link The email sign-in link.
        @return @YES when the link passed matches the expected format of an email sign-in link.
     */
    open func isSignIn(withEmailLink link: String) -> Bool

That's the only function I found. Im not sure if thats what you meant?

Reply:
maybe [this](https://stackoverflow.com/questions/37738366/how-to-check-if-user-has-valid-auth-session-firebase-ios) but dont think it matters too much haha, sorry